extensions/procfs/processors/ProcFsMonitor.h

```diff
@@ -0,0 +1,135 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <string>
+#include <memory>
+#include <map>
+#include <vector>
+#include <utility>
+#include <optional>
+
+#include "../ProcFs.h"
+#include "core/Processor.h"
+#include "core/logging/LoggerConfiguration.h"
+#include "core/logging/Logger.h"
+#include "utils/Enum.h"
+
+#include "rapidjson/stream.h"
+#include "rapidjson/document.h"
+
+namespace org::apache::nifi::minifi::extensions::procfs {
+
+class ProcFsMonitor : public core::Processor {
+ public:
+  explicit ProcFsMonitor(const std::string& name, utils::Identifier uuid = utils::Identifier())
+      : Processor(name, uuid) {
+  }
+  ProcFsMonitor(const ProcFsMonitor&) = delete;
+  ProcFsMonitor(ProcFsMonitor&&) = delete;
+  ProcFsMonitor& operator=(const ProcFsMonitor&) = delete;
+  ProcFsMonitor& operator=(ProcFsMonitor&&) = delete;
+  ~ProcFsMonitor() override = default;
```

Comment:
I think the base class CoreComponent has these members deleted

Reply:
thanks that's good to know, less boilerplate
removed them in https://github.com/apache/nifi-minifi-cpp/pull/1152/commits/df0ddde8ad2fa559dee75a3c51752a6e0d2f0eb0